cmd/testwrapper: test flake in test flake wrapper tests?

```
?	tailscale.com/cmd/sync-containers [skipped/no tests] 
?	tailscale.com/cmd/tailscale [skipped/no tests] 
ok	tailscale.com/cmd/tailscale/cli
?	tailscale.com/cmd/tailscaled/childproc [skipped/no tests] 
?	tailscale.com/cmd/testcontrol [skipped/no tests] 
ok	tailscale.com/cmd/tailscaled
=== RUN   TestFlakeRun
    flakytest.go:38: flakytest: issue tracking this flaky test: https://github.com/tailscale/tailscale/issues/0
    flakytest_test.go:41: First run in testwrapper, failing so that test is retried. This is expected.
--- FAIL: TestFlakeRun (0.00s)
FAIL	tailscale.com/cmd/testwrapper/flakytest
?	tailscale.com/cmd/tsconnect [skipped/no tests] 
=== RUN   TestBuildError
=== PAUSE TestBuildError
=== CONT  TestBuildError
    testwrapper_test.go:167: testwrapper /tmp/TestBuildError193629[75](https://github.com/tailscale/tailscale/actions/runs/6137434026/job/16653250254?pr=9325#step:8:76)00/001/builderror_test.go: expected error with exit code 0 but got: <nil>
--- FAIL: TestBuildError (0.42s)
?	tailscale.com/cmd/viewer [skipped/no tests] 
?	tailscale.com/cmd/viewer/tests [skipped/no tests] 
ok	tailscale.com/control/controlbase
FAIL	tailscale.com/cmd/testwrapper
ok	tailscale.com/control/controlclient
?	tailscale.com/control/controlknobs [skipped/no tests]
```

/cc @maisem @icio

--- a/cmd/testwrapper/testwrapper.go
+++ b/cmd/testwrapper/testwrapper.go
@@ -89,11 +89,6 @@ func runTests(ctx context.Context, attempt int, pt *packageTests, otherArgs []st
 		log.Printf("error starting test: %v", err)
 		os.Exit(1)
 	}
-	cmdErr := make(chan error, 1)
-	go func() {
-		defer close(cmdErr)
-		cmdErr <- cmd.Wait()
-	}()
 
 	s := bufio.NewScanner(r)
 	resultMap := make(map[string]map[string]*testAttempt) // pkg -> test -> testAttempt
@@ -164,10 +159,13 @@ func runTests(ctx context.Context, attempt int, pt *packageTests, otherArgs []st
 			}
 		}
 	}
-	if err := s.Err(); err != nil {
+	if err := cmd.Wait(); err != nil {
 		return err
 	}
-	return <-cmdErr
+	if err := s.Err(); err != nil {
+		return fmt.Errorf("reading go test stdout: %w", err)
+	}
+	return nil
 }
 
 func main() {
@@ -303,6 +301,8 @@ func main() {
 						os.Exit(exit.ExitCode())
 					}
 				}
+				log.Printf("testwrapper: %s", err)
+				os.Exit(1)
 			}
 		}
 		if len(toRetry) == 0 {

